scripts/gitOpsApi.js

```diff
@@ -0,0 +1,429 @@
+	 */
+	updateSingleFile: async (octokit, options) => {
+	    return new Promise((resolve, reject) => {
+	        octokit.request("PUT /repos/{owner}/{repo}/contents/{path}", {
+	            owner: options.repoOwner,
+	            repo: options.repoName,
+	            path: options.filePath,
+	            message: options.commitMsg,
+	            content: Buffer.from(options.fileContent).toString("base64"),
+	            branch: options.branchName,
+				sha: options.sha
+	        }).then(() => {
+	            resolve(options.filePath + " has been updated successfully.");
+	        }).catch((e) => {
+	            reject({
+					isError: true,
+					message: "Error at updateSingleFile(): " + e.message
+				});
+	        });
+	    });
+	},
+
+	/**
+	 * An object that contains required information for submitting multiple files in one commit.
+	 * Note that when creating or updating a single file, API functions createSingleFile() and updateSingleFile()
+	 * are recommended for better performance.
+	 * @typedef {Object} CommitMultipleFilesOptions
+	 * @param {String} repoOwner - The repo owner.
+	 * @param {String} repoName - The repo name.
+	 * @param {String} branchName - The name of the remote branch to operate.
+	 * @param {Object[]} files - An array of objects. Each object contains a file information in a structure of
+	 * [{path: {String}, content: {String}}, ...].
+	 * @param {String} commitMsg - The commit message.
+	 */
+
+	 /**
+ 	 * Submit multiple files in one commit.
+ 	 * @param {Object} octokit - An instance of octokit with authentication being set.
+ 	 * @param {CommitMultipleFilesOptions} options - Required parameters for this github operation.
+ 	 * @return {Promise} The resolve or reject of the promise indicates whether the operation completes successfully.
+ 	 */
+	commitMultipleFiles: async (octokit, options) => {
+		let parentCommit;
+		const repoOwner = options.repoOwner;
+		const repoName = options.repoName;
+		const branchName = options.branchName;
+		const files = options.files;
+		const commitMsg = options.commitMsg;
+
+		/**
+		 * Returns a single tree using the SHA1 value for that tree.
+		 * @param {String} repoOwner - The repo owner.
+		 * @param {String} repoName - The repo name.
+		 * @param {String} branchName - The name of the remote branch to get the tree from.
+		 * @return {Promise} The resolve or reject of the promise indicates whether the operation completes successfully.
+		 */
+		const getTree = async function (repoOwner, repoName, branchName) {
+			return module.exports.getBranchRef(octokit, {
+				repoOwner: repoOwner,
+				repoName: repoName,
+				branchName: branchName
+			}).then((commit) => {
+				parentCommit = commit;
+				return octokit.request("GET /repos/{owner}/{repo}/git/trees/{tree_sha}", {
+					owner: repoOwner,
+					repo: repoName,
+					tree_sha: commit.object.sha
+				});
+			});
+		};
+
+		return new Promise((resolve, reject) => {
```

Comment:
This is quite a complex sequence of actions - it would be helpful to include a link to either documentation or sample code from which it was determined that this is the right sequence

Reply:
Bump comment - any source for this sequence or did you come up with it just for this API?